service/src/main/java/com/zebrunner/reporting/service/TestRunArtifactService.java

```diff
@@ -21,18 +24,21 @@ public TestRunArtifactService(TestRunArtifactMapper testRunArtifactMapper) {
     }
 
     @Transactional
-    public TestRunArtifact createTestRunArtifact(TestRunArtifact testRunArtifact) {
-        TestRunArtifact dbTestRunArtifact = getTestRunArtifactByNameAndTestRunId(testRunArtifact.getName(), testRunArtifact.getTestRunId());
-        if (dbTestRunArtifact != null) {
-            throw new IllegalOperationException(TEST_RUN_ARTIFACT_CAN_NOT_BE_CREATED, String.format(TEST_RUN_ARTIFACT_ALREADY_EXISTS, dbTestRunArtifact.getName()));
+    public List<TestRunArtifact> createTestRunArtifacts(List<TestRunArtifact> testRunArtifacts) {
+        List<String> alreadyExistingNames = testRunArtifacts.stream()
+                                                            .filter(artifact -> existsByNameAndTestRunId(artifact.getName(), artifact.getTestRunId()))
+                                                            .map(TestRunArtifact::getName)
+                                                            .collect(Collectors.toList());
+        if (!alreadyExistingNames.isEmpty()) {
+            String names = String.join(", ", alreadyExistingNames);
```

Comment:
Minor note. It seems like you can use `.collect(Collectors.joining(", "));` two lines above in order to not create the `alreadyExistingNames` list. The `if` clause can be replaced by checking the resulting `String` with `#isEmpty()`.

Reply:
@SergeyBrenko are you sure that it is a good idea to check string length instead of list size? (list size validation looks nice)

Reply:
Actually, I am fine with both solutions. `List` looks more traditional, but requires more RAM and CPU time. On the other hand, the pre-formatted `String` looks a bit weird, but it works faster and takes up less memory without any side effect (if i didn't miss something). Anyway, it is up to you to choose the solution.